clients/src/test/java/org/apache/kafka/clients/NetworkClientApiVersionsCheckTest.java

```diff
@@ -0,0 +1,108 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients;
+
+
+import org.apache.kafka.common.KafkaException;
+import org.apache.kafka.common.network.NetworkReceive;
+import org.apache.kafka.common.protocol.ApiKeys;
+import org.apache.kafka.common.protocol.ProtoUtils;
+import org.apache.kafka.common.protocol.Protocol;
+import org.apache.kafka.common.protocol.types.Struct;
+import org.apache.kafka.common.requests.ApiVersionsResponse;
+import org.apache.kafka.common.requests.ResponseHeader;
+import org.apache.kafka.test.DelayedReceive;
+import org.junit.Test;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+import static org.apache.kafka.common.requests.ApiVersionsResponse.API_KEY_NAME;
+import static org.apache.kafka.common.requests.ApiVersionsResponse.API_VERSIONS_KEY_NAME;
+import static org.apache.kafka.common.requests.ApiVersionsResponse.ERROR_CODE_KEY_NAME;
+import static org.apache.kafka.common.requests.ApiVersionsResponse.MAX_VERSION_KEY_NAME;
+import static org.apache.kafka.common.requests.ApiVersionsResponse.MIN_VERSION_KEY_NAME;
+import static org.junit.Assert.fail;
+
+public class NetworkClientApiVersionsCheckTest extends NetworkClientTest {
+
+    private static final List<Short> USED_API_KEYS = Arrays.asList(
+            ApiKeys.METADATA.id,
+            ApiKeys.FETCH.id,
+            ApiKeys.GROUP_COORDINATOR.id,
+            ApiKeys.HEARTBEAT.id,
+            ApiKeys.JOIN_GROUP.id,
+            ApiKeys.LEAVE_GROUP.id,
+            ApiKeys.LIST_OFFSETS.id,
+            ApiKeys.OFFSET_COMMIT.id,
+            ApiKeys.OFFSET_FETCH.id,
+            ApiKeys.SYNC_GROUP.id);
+
+    @Override
+    protected Collection<ApiVersionsResponse.ApiVersion> getExpectedApiVersions() {
+        List<ApiVersionsResponse.ApiVersion> expectedApiVersions = new ArrayList<>();
+        for (Short apiKey: this.USED_API_KEYS)
+            expectedApiVersions.add(new ApiVersionsResponse.ApiVersion(apiKey, Protocol.MIN_VERSIONS[apiKey], Protocol.CURR_VERSION[apiKey]));
+        return expectedApiVersions;
+    }
+
+    @Test
+    public void testUnsupportedLesserApiVersions() {
+        unsupportedApiVersionsCheck(getExpectedApiVersions(), (short) (Short.MAX_VALUE - 2), Short.MAX_VALUE);
```

Comment:
Not sure if those changes help, but making those changes anyway. Let me know if you meant something else.